### PR TITLE
Setup cards backend

### PR DESCRIPTION
**Pull request by ryutaudo**
## Implementation
- organize file structure in the backend
- migration file for flashcards
- seed file for flashcards and users
- API server for flashcards
- Test for API server for flashcards

### CODE REVIEW

File: server/__tests__/routes/flashcards.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+const PORT = process.env.PORT || 9999;
+const END_POINT = '/api/users/1/flashcards';
+
+const server = app.listen(PORT);
+chai.use(chaiHttp);
+
+describe('server/routes/flashcards', () => {
+  // afterAll((done) => {
+  //   chai.request(server).server.close(done);
```
Comment (ChristianSchmidt1981) [inline server/__tests__/routes/flashcards.js]: please delete this code

File: server/__tests__/routes/flashcards.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        });
+    });
+  });
+
+  describe('DELETE /users/:userId/flashcards/:id', () => {
+    it('should delete data of a flashcard', (done) => {
+      chai.request(server)
+        .delete(`${END_POINT}/1`)
+        .set('Content-Type', 'application/json')
+        .end((err, res) => {
```
Comment (ChristianSchmidt1981) [inline server/__tests__/routes/flashcards.js]: please rename err to error
res to response

File: server/db/flashcards/create.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module.exports = knex => async (params) => {
+  const flashcard = params;
```
Comment (ChristianSchmidt1981) [inline server/db/flashcards/create.js]: please rename params to flashcards. than you don't need extra an additional const

File: server/db/flashcards/get.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module.exports = knex => async (params) => {
+  const id = params;
```
Comment (ChristianSchmidt1981) [inline server/db/flashcards/get.js]: please rename params to id and remove the const. this is not more needed

File: server/db/flashcards/list.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+module.exports = knex => async (params) => {
+  const userId = params;
```
Comment (ChristianSchmidt1981) [inline server/db/flashcards/list.js]: please rename params to userId and remove the not needed const

File: server/db/flashcards/remove.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+module.exports = knex => async (params) => {
+  const id = params;
```
Comment (ChristianSchmidt1981) [inline server/db/flashcards/remove.js]: please rename params to id and remove the not needed const

File: server/db/users/addUser.js [hunk 7]
```diff
@@ -1,15 +1,14 @@
...
-        name: user.name,
-        email: user.email,
-        password: user.password,
-        created_at: user.created_at
-      });
-    } catch (error) {
-      return error;
-    }
+module.exports = knex => async (params) => {
+  const user = params;
```
Comment (ChristianSchmidt1981) [inline server/db/users/addUser.js]: please rename params to user and remove the not needed const

File: server/db/users/findById.js [hunk 8]
```diff
@@ -1,10 +1,9 @@
-module.exports = knex => {
-  return async params => {
-    const id = params;
-    try {
-      return await knex('users').where('id', id);
-    } catch (error) {
-      return error;
-    }
+module.exports = knex => async (params) => {
+  const id = params;
```
Comment (ChristianSchmidt1981) [inline server/db/users/findById.js]: please rename params to id and remove the not needed const

File: server/routes/flashcards.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+/* GET all the flashcards. */
+router.get('/users/:userId/flashcards', async (req, res) => {
+  const { userId } = req.params;
+  const flashcards = await db.flashcards.list(userId);
+  res.json(flashcards);
+});
+
+/* POST a flashcard. */
+router.post('/users/:userId/flashcards', (req, res) => {
```
Comment (ChristianSchmidt1981) [inline server/routes/flashcards.js]: rename req to request
res to response

File: server/routes/flashcards.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+      userId,
+      status: 'success',
+    });
+  } catch (error) {
+    res.status(500).send(error);
+  }
+});
+
+/* GET a flashcard. */
+router.get('/users/:userId/flashcards/:id', async (req, res) => {
```
Comment (ChristianSchmidt1981) [inline server/routes/flashcards.js]: rename req to request
res to response

File: server/routes/flashcards.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  const { id } = req.params;
+  const flashcard = await db.flashcards.get(id);
+  res.json(flashcard);
+});
+
+/* DELETE a flashcard. */
+router.delete('/users/:userId/flashcards/:id', (req, res) => {
```
Comment (ChristianSchmidt1981) [inline server/routes/flashcards.js]: rename req to request
res to response

**Review by ChristianSchmidt1981 — APPROVED** [review]
please rename some variables and remove not needed code

File: server/db/migrations/20180314111434_add_users_table.js [hunk 12]
```diff
@@ -1,23 +1,27 @@
...
       .defaultTo(knex.fn.now());
-  });
-};
+  })
+);
 
-exports.down = (knex, Promise) => {
-  return knex.schema.dropTableIfExists('users');
-};
+exports.down = knex => (
```
Comment (ChristianSchmidt1981) [inline server/db/migrations/20180314111434_add_users_table.js]: please remove () breakets

File: server/db/migrations/20180315115126_add_flashcards_table.js [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+      .notNullable()
+      .index();
+
+    table.timestamp('created_at')
+      .notNullable()
+      .defaultTo(knex.fn.now());
+  })
+);
+
+exports.down = knex => (
```
Comment (ChristianSchmidt1981) [inline server/db/migrations/20180315115126_add_flashcards_table.js]: please remove () breakets

**Review by ChristianSchmidt1981 — APPROVED** [review]
please remove some breakets

**Review by ChristianSchmidt1981 — APPROVED** [review]
i like it

File: server/db/knexfile.js [hunk 14]
```diff
@@ -1,37 +1,53 @@
...
+    },
+    migrations: {
+      tableName: 'knex_migrations',
+      directory: `${__dirname}/migrations`,
+    },
+    seeds: {
+      directory: `${__dirname}/seed/`,
+    },
+  },
+  test: {
```
Comment (okasato) [inline server/db/knexfile.js:37]: In my code, I don't use test environment. I use staging environment.

File: server/db/users/findByAccount.js [hunk 15]
```diff
@@ -1,10 +1,8 @@
...
-  return async params => {
-    const account = params;
-    try {
-      return await knex('users').where('email', params);
-    } catch (error) {
-      return error;
-    }
+module.exports = knex => async (acount) => {
+  try {
+    return await knex('users').where('email');
```
Comment (okasato) [inline server/db/users/findByAccount.js]: Why did you remove params in knex('users').where('email', params)?

**Review by okasato — APPROVED** [review]
